oauth2/oauth2-resource-server/src/test/java/org/springframework/security/oauth2/server/resource/web/DefaultBearerTokenResolverTests.java

```diff
@@ -153,8 +213,22 @@ public void resolveWhenQueryParameterIsPresentAndSupportedThenTokenIsResolved()
 	public void resolveWhenQueryParameterIsPresentAndNotSupportedThenTokenIsNotResolved() {
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		request.setMethod("GET");
+		request.setQueryString("access_token=" + TEST_TOKEN);
 		request.addParameter("access_token", TEST_TOKEN);
 
 		assertThat(this.resolver.resolve(request)).isNull();
 	}
+
+	@Test
+	public void resolveWhenQueryParameterIsAbsentAndSupportedWithFormParameterIsPresentAndNotSupportedThenTokenIsResolved() {
+		this.resolver.setAllowUriQueryParameter(true);
+
+		MockHttpServletRequest request = new MockHttpServletRequest();
+		request.setMethod("POST");
+		request.setContentType("application/x-www-form-urlencoded");
+		request.addParameter("access_token", TEST_TOKEN);
+
+		assertThat(this.resolver.resolve(request)).isEqualTo(TEST_TOKEN);
```

Comment:
Since form parameter would be more secure than query parameter, I did not add that additional check hence it will resolve. Let me know if this needs to change.

Reply:
You are making a significant behavior change to `DefaultBearerTokenResolver` here - the point of `allowFormEncodedBodyParameter` switch is to enable form-encoded body parameter and you're making it implicitly enabled via `allowUriQueryParameter` here without that being documented anywhere.

Reply:
This is how even the existing behavior is. But I do agree with your comment and I will update it as suggested.

Reply:
@vpavic, **please let me know what you think.**
I was making change as "proposed by @vpavic" to "current test" but realized that "current test" is much similar to as "proposed by @gburboz" (note words Form and Query are only swapped). These last two tests should have similar behaviour is what I am suggesting (could be either `null` or `AuthenticationException`).

- `resolveWhenFormParameterIsPresentAndSupportedWithInvalidContentTypeThenTokenIsNotResolved` (**proposed by @vpavic** )
- `resolveWhenQueryParameterIsAbsentAndSupportedWithFormParameterIsPresentAndNotSupportedThenXyzResult` (**current test**)
- `resolveWhenFormParameterIsAbsentAndSupportedWithQueryParameterIsPresentAndNotSupportedThenAuthenticationExceptionIsThrown` (**proposed by @gburboz** )

Reply:
As proposed by  @vpavic, I have updated test case to `resolveWhenFormParameterIsPresentAndSupportedWithInvalidContentTypeThenTokenIsNotResolved`